File: mpp-proiect-repository-cristinahendre/lab8(bun)/Java/lab6/modul/RepoModule/src/main/java/Repository/CazRepo.java
package Repository;

import Domain.CazCaritabil;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;


import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;


public class CazRepo implements  CazRepoI {

    private  JdbcUtils utils;

    public CazRepo(Properties prop) {
        logger.info("intializare cu proprietati {}", prop);
        utils= new JdbcUtils(prop);
    }

    private  static  final Logger logger = LogManager.getLogger();

    @Override
    public CazCaritabil findOne(Integer integer) {

        logger.traceEntry("Caut cazul caritabil cu acest id: {}", integer);

        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement("select * from caz where id = ?")){

            statement.setInt(1,integer);
            try(ResultSet resultSet = statement.executeQuery()){
                if(resultSet.next()){
                    int id = resultSet.getInt(1);
                    String nume= resultSet.getString(2);
                    int suma =resultSet.getInt(3);
                    CazCaritabil caz =new CazCaritabil(nume);
                    caz.setSuma_donata(suma);
                    caz.setId(id);
                    return caz;
                }
            }

            con.close();

        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;

    }

    @Override
    public Iterable<CazCaritabil> findAll() {

        logger.traceEntry("afisez toate elementele ");
        List<CazCaritabil> lista = new ArrayList<>();
        Connection con = utils.getConnection();

        try(PreparedStatement pst = con.prepareStatement("select * from caz")){

            try(ResultSet resultSet = pst.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");
                    String nume= resultSet.getString("nume");
                    int suma = resultSet.getInt(3);

                    CazCaritabil c= new CazCaritabil(nume);
                    c.setSuma_donata(suma);
                    c.setId(id);
                    lista.add(c);
                }
            }
            //con.close();
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }

        return  lista;
    }

    @Override
    public CazCaritabil save(CazCaritabil entity) {

        logger.traceEntry("Adaug cazul caritabil {}", entity);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("insert into caz(nume,suma_donata) values (?,?)")){


            statement.setString(1,entity.getNume());
            statement.setInt(2,entity.getSuma_donata());

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            con.close();
            return entity;



        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return  null;
    }

    @Override
    public void delete(Integer integer) {

        logger.traceEntry("Sterg cazul caritabil cu id-ul {}", integer);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("delete from caz where id = ?")){

            statement.setInt(1,integer);

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            con.close();




        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }

    }

    @Override
    public CazCaritabil update(CazCaritabil entity) {

        logger.traceEntry("Modific cazul caritabil {}", entity);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("update caz set nume=? , suma_donata =? where id =?")){

            statement.setString(1,entity.getNume());
            statement.setInt(2,entity.getSuma_donata());
            statement.setInt(3,entity.getId());

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            //con.close();
            return entity;



        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return  null;
    }

    @Override
    public int nrElem() {
        logger.traceEntry("Numar cate elemente sunt in caz");
        int nr =0;
        Connection con = utils.getConnection();
        try(Statement statement = con.createStatement()){

            ResultSet resultSet = statement.executeQuery("select count(*) from caz");
            resultSet.next();
            nr= resultSet.getInt(1);
            con.close();
            return nr;


        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        logger.traceExit(nr);
        return nr;
    }
}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Repository/CazRepo.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Cerinta1.Domain;
using Cerinta1.Domain.Validator;
using log4net;
using log4net.Config;

namespace Cerinta1.Repository
{
  
    class CazRepo : ICazRepo

    {
     
        private static readonly ILog log = LogManager.GetLogger(nameof(CazRepo));


        public CazCaritabil Delete(int id)
        {
            IDbConnection con = DBUtils.getConnection();
            log.InfoFormat("incerc sa sterg cazul caritabil {0}", id);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from caz where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                con.Close();
                if (dataR == 0)
                {
                    log.InfoFormat("Eroare la stergerea cazului.");
                    throw new RepositoryException("Nu s-a sters cu succes!");
                }
                else
                {
                    log.InfoFormat("Stergerea cazului a avut loc cu succes.");
                    return null;
                }
            }
        }


        public IEnumerable<CazCaritabil> FindAll()
        {
            IList<CazCaritabil> lista = new List<CazCaritabil>();
            log.InfoFormat("Afisez toate cazurile caritabile");
            IDbConnection con = DBUtils.getConnection();
            try
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select * from caz ";
                   

                    using (var dataR = comm.ExecuteReader())
                    {
                        while (dataR.Read())
                        {
                            int idV = dataR.GetInt32(0);
                            string nume = dataR.GetString(1);
                            int suma_donata = dataR.GetInt32(2);
                            CazCaritabil caz = new CazCaritabil(nume, suma_donata);
                            caz.Id = idV;
                           
                            lista.Add(caz);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            con.Close();
            log.InfoFormat("Am parcurs.");
            return lista;
        }




        public CazCaritabil FindOne(int id)
        {
            log.Info("Caut cazul cu acest ");
            IDbConnection con = DBUtils.getConnection();
            try
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select * from caz where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);

                    using (var dataR = comm.ExecuteReader())
                    {
                        if (dataR.Read())
                        {
                            int idV = dataR.GetInt32(0);
                            string nume = dataR.GetString(1);
                            int suma_donata = dataR.GetInt32(2);
                            CazCaritabil caz = new CazCaritabil(nume, suma_donata);
                            caz.Id = idV;
                            log.InfoFormat("am gasit cazul {0}", caz);
                            con.Close();
                            return caz;
                        }
                    }
                }
            }
            catch(Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            log.InfoFormat("nu am  gasit acel caz.");
            return null;
        }



        public int nrElem()
        {
            log.InfoFormat("Numar cate cazuri sunt in baza de date.");
            IDbConnection con = DBUtils.getConnection();
            try
            {
                using(var comm = con.CreateCommand())
                {
                    comm.CommandText = "select count(*) from caz";
                    int rez = int.Parse(comm.ExecuteScalar().ToString());
                    con.Close();
                    log.InfoFormat("Sunt {0} cazuri.", rez);
                    return rez;
                }


            }
            catch(Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return -1;
        }



        public CazCaritabil Save(CazCaritabil entity)
        {
            log.InfoFormat("Voi salva cazul caritabil {0}",entity);
            try
            {
                IDbConnection con = DBUtils.getConnection();
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "insert into caz(nume,suma_donata) values (@nume,@s)";
                    


                    IDbDataParameter paramNume = comm.CreateParameter();
                    paramNume.ParameterName = "@nume";
                    paramNume.Value = entity.Nume;
                    comm.Parameters.Add(paramNume);

                    IDbDataParameter paramSuma = comm.CreateParameter();
                    paramSuma.ParameterName = "@s";
                    paramSuma.Value = entity.Suma_donata;


                    comm.Parameters.Add(paramSuma);

                    var dataR = comm.ExecuteNonQuery();
                    con.Close();
                    if (dataR == 0)
                    {
                        log.InfoFormat("Salvarea cazului a intampinat probleme");
                        throw new RepositoryException("Nu s-a adaugat!");
                    }
                    else
                    {
                        log.InfoFormat("Salvarea cazului e cu succes.");
                        return entity;
                    }
                }
            }
            catch(Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return null;
        }



        public CazCaritabil Update(CazCaritabil entity)
        {
            log.InfoFormat("Modific cazul {0}",entity);
            try
            {
                IDbConnection con = DBUtils.getConnection();
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "update caz set nume=@nume, suma_donata=@s where id = @id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = entity.Id;
                   


                    IDbDataParameter paramNume = comm.CreateParameter();
                    paramNume.ParameterName = "@nume";
                    paramNume.Value = entity.Nume;
                    comm.Parameters.Add(paramNume);

                    IDbDataParameter paramSuma = comm.CreateParameter();
                    paramSuma.ParameterName = "@s";
                    paramSuma.Value = entity.Suma_donata;


                    comm.Parameters.Add(paramSuma);
                    comm.Parameters.Add(paramId);

                    var dataR = comm.ExecuteNonQuery();
                    con.Close();
                    if (dataR == 0)
                    {
                        log.InfoFormat("Eroare la modificarea cazului.");
                        throw new RepositoryException("Nu s-a modificat!");
                    }
                    else {
                        log.InfoFormat("Caz modificat cu succes.");
                        return entity; }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return null;
        }
    }
}
